java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -7899,6 +7900,59 @@ void testParquetWriteToFileUncompressedNoStats() throws IOException {
     }
   }
 
+  @Test
+  void testParquetWriteWithFieldId() throws IOException {
+    // field IDs are:
+    // c1: -1, c2: 2, c3: 3, c31: 31, c32: 32, c4: -4, c5: not specified
+    ColumnWriterOptions.StructBuilder sBuilder =
+        structBuilder("c3", true, 3)
+            .withColumn(true, "c31", 31)
+            .withColumn(true, "c32", 32);
```

Comment:
Would be good to add a negative test for a nested column, as it's a slightly different code path that appears to be problematic in the current version of this PR.

Reply:
Added a test case.

Spark allows two columns have identical `parquet.field.id`, do did not test this kind of case.

```
from pyspark.sql.types import *
schema = StructType([
    StructField("c1", IntegerType(), metadata={'parquet.field.id' : 1}),
    StructField("c2", IntegerType(), metadata={'parquet.field.id' : 1})
])
data = [(1,2,),(11,22,),(33,33,),]
spark.createDataFrame(data, schema).coalesce(1).write.mode("overwrite").parquet("/tmp/tmp-parquet")
```

parquet-tool meta /tmp/tmp-parquet/xxx.parquet

```
......
Schema:
message spark_schema {
  optional int32 c1 = 1;
  optional int32 c2 = 1;
}
```